lib/services/provided/autocomplete.js

```diff
@@ -145,19 +146,32 @@ export function provideAutocompleteResults(
       const kernel = store.kernel;
       if (!kernel || kernel.executionState !== "idle") return null;
 
-      return new Promise(resolve => {
-        kernel.inspect(replacedText, replacedText.length, ({ data, found }) => {
-          if (!found || !data["text/plain"]) resolve(null);
-          resolve({
+      const promise = new Promise(resolve => {
+        kernel.inspect(replacedText, replacedText.length, ({ found, data }) => {
+          if (!found || !data["text/plain"]) {
+            return resolve(null);
+          }
+          const description = ansiToText(data["text/plain"]);
+          return resolve({
             text,
             replacementPrefix,
             replacedText,
             iconHTML,
             type,
-            description: ansiToText(data["text/plain"])
+            description
           });
         });
       });
+
+      return Promise.race([promise, this.timeout()]);
+    },
+
+    timeout() {
```

Comment:
`timeout` and `Promise.race` is introduced here just because it would make autocompletions smoother when a kernel is likely to be busy (e.g. just after starting up or when with experimental description completions)